src/net/mod.rs

```diff
@@ -69,9 +69,16 @@ impl Connector {
 
     pub async fn connect(&self, addr: &str) -> Result<Stream> {
         match &self.inner {
-            ConnectorImpl::Tcp => Ok(Stream {
-                inner: StreamImpl::Tcp(tokio::net::TcpStream::connect(addr).await?),
-            }),
+            ConnectorImpl::Tcp => {
+                let s = tokio::net::TcpStream::connect(addr).await?;
+                let res = s.set_nodelay(true);
+                if res.is_err() {
+                    eprintln!("Error setting TCP_NODELAY: {:?}", res.err());
+                }
```

Comment:
I realize this is already merged but in the future it would be cleaner to do
```rs
if let Err(e) = s.set_nodelay(true) {
    eprintln!("Error setting TCP_NODELAY: {e:?}");
}
```